test/e2e/metrics_e2e_test.go

```diff
@@ -4,88 +4,116 @@ package e2e
-
-		failingCSV := v1alpha1.ClusterServiceVersion{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       v1alpha1.ClusterServiceVersionKind,
-				APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name: genName("failing-csv-test-"),
-			},
-			Spec: v1alpha1.ClusterServiceVersionSpec{
-				InstallStrategy: v1alpha1.NamedInstallStrategy{
-					StrategyName: v1alpha1.InstallStrategyNameDeployment,
-					StrategySpec: strategy,
-				},
-			},
-		}
+var _ = Describe("Metrics are generated for OLM pod", func() {
 
-		cleanupCSV, err := createCSV(GinkgoT(), c, crc, failingCSV, testNamespace, false, false)
-		require.NoError(GinkgoT(), err)
-		defer cleanupCSV()
+	var (
+		c   operatorclient.ClientInterface
+		crc versioned.Interface
+	)
 
-		_, err = fetchCSV(GinkgoT(), crc, failingCSV.Name, testNamespace, csvFailedChecker)
-		require.NoError(GinkgoT(), err)
+	BeforeEach(func() {
+		c = newKubeClient(GinkgoT())
+		crc = newCRClient(GinkgoT())
 
-		rawOutput, err := getMetricsFromPod(GinkgoT(), c, getOLMPodName(GinkgoT(), c), operatorNamespace, "8081")
-		if err != nil {
-			GinkgoT().Fatalf("Metrics test failed: %v\n", err)
-		}
+	})
 
-		// Verify metrics have been emitted for packageserver csv
-		require.Contains(GinkgoT(), rawOutput, "csv_abnormal")
-		require.Contains(GinkgoT(), rawOutput, "name=\""+failingCSV.Name+"\"")
-		require.Contains(GinkgoT(), rawOutput, "phase=\"Failed\"")
-		require.Contains(GinkgoT(), rawOutput, "reason=\"UnsupportedOperatorGroup\"")
-		require.Contains(GinkgoT(), rawOutput, "version=\"0.0.0\"")
+	Context("Given an operator group that supports all namespaces", func() {
+		When("a CSV spec does not include Install Mode", func() {
+
+			var (
+				cleanupCSV cleanupFunc
+				failingCSV v1alpha1.ClusterServiceVersion
+			)
+
+			BeforeEach(func() {
+
+				failingCSV = v1alpha1.ClusterServiceVersion{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       v1alpha1.ClusterServiceVersionKind,
+						APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: genName("failing-csv-test-"),
+					},
+					Spec: v1alpha1.ClusterServiceVersionSpec{
+						InstallStrategy: v1alpha1.NamedInstallStrategy{
+							StrategyName: v1alpha1.InstallStrategyNameDeployment,
+							StrategySpec: strategy,
+						},
+					},
+				}
+
+				var err error
+				cleanupCSV, err = createCSV(c, crc, failingCSV, testNamespace, false, false)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = fetchCSV(crc, failingCSV.Name, testNamespace, csvFailedChecker)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				if cleanupCSV != nil {
+					cleanupCSV()
+				}
+			})
+			It("CSV will transition to a failure state with reason UnsupportedOperatorGroup", func() {
+
+				// TestMetrics tests the metrics endpoint of the OLM pod.
+				rawOutput, err := getMetricsFromPod(c, getOLMPodName(c), operatorNamespace, "8081")
+				if err != nil {
```

Comment:
`Expect(err).NotTo(HaveOccurred())` should accomplish the same thing as the old `*T.Fatalf` behavior.